### PR TITLE
Patron: Adds a new animated welcome view

### PR DESCRIPTION
**Pull request by emilylaguna**
Sorry about the size of this PR.

This adds a new animated welcome view for Patron's that introduces the new app icons in a fun manner along with some custom haptics for a slam, spring, and unlocking rumble effect.

The slam and spring effects are [AHAP files](https://developer.apple.com/documentation/corehaptics/representing_haptic_patterns_in_ahap_files) and the unlock rumble effect dynamically made in code using CoreHaptics.

## Demo Videos

<details>
  <summary><strong>iPhone 14</strong></summary>
  
https://github.com/Automattic/pocket-casts-ios/assets/793774/f38abbdc-9e58-4e02-b86e-9c5c72e5a266

</details>

<details>
  <summary><strong>iPod Touch</strong></summary>

https://github.com/Automattic/pocket-casts-ios/assets/793774/47080bf1-5747-40a7-90e1-0c26c33fab17

</details>

<details>
  <summary><strong>iPad</strong></summary>

https://github.com/Automattic/pocket-casts-ios/assets/793774/b6935954-f7bd-419e-8046-92b2ab4b220f

</details>

<details>
  <summary><strong>Reduced Motion Enabled - Animations disabled</strong></summary>
  
https://github.com/Automattic/pocket-casts-ios/assets/793774/e632893a-daa7-49ec-9928-28611d03d8bf

</details>

## To test

**📱 Note:** You should test this on a real device.

Please also test on smaller devices, iOS 15, and iPad.

### Welcome View for Patron's

1. Enable the StoreKit config by: Hitting Command-Shift-Comma in Xcode, and changing the 'StoreKit Config' to the Pocket Casts config
2. Enable the Patron feature flag if it's not already enabled
3. Open the upgrade view from anywhere (such as profile > account)
4. Select Patron and purchase it
5. ✅ Verify you are brought to the new welcome view
6. Tap and hold the unlock button and stop before it completes
7. ✅ Verify the progress resets
8. Tap and hold the unlock button
9. ✅ Verify the pinwheel background slow fades in
10. Release the unlock button once it's at 100%
11. ✅ Verify the icons appear and a haptic plays for each one that lands
12. ✅ Verify the view looks good
13. Tap on any icon
14. ✅ Verify it becomes selected and you see the a back button and a change button
15. Rotate your device
16. ✅ Verify the app icon also moves and there is a glossy shine effect
17. Tap back
18. ✅ Verify you go back to the icons view
19. Tap another icon
20. Tap the Change App Icon button
21. ✅ Verify you see an alert that the app icon was changed
22. Tap the Continue button
23. ✅ Verify the view is dismissed

### Welcome view for Plus

1. Use the Developer Menu to 'Set to No Plus'
2. Open the upsell view again
3. Subscribe to Plus
4. ✅ Verify you are brought to the old welcome view

### Feature Flag disabled
1. Use the Developer Menu to 'Set to No Plus'
2. Disable the patron flag
3. Open the upsell view again
4. Subscribe to Plus
5. ✅ Verify you are brought to the old welcome view

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/Haptics/Slam.ahap
Comment (emilylaguna) [inline podcasts/Haptics/Slam.ahap:1]: If you're interested in what this looks like, you can open the file in Finder and use QuickLook to view it:

File: podcasts/Haptics/Spring.ahap
Comment (emilylaguna) [inline podcasts/Haptics/Spring.ahap:1]: If you're interested in what this looks like, you can open the file in Finder and use QuickLook to view it:

File: podcasts/Onboarding/Models/PlusPurchaseModel.swift [hunk 1]
```diff
@@ -119,7 +119,13 @@ private extension PlusPurchaseModel {
         guard let parentController else { return }
 
         let navigationController = parentController as? UINavigationController
-        let controller = WelcomeViewModel.make(in: navigationController, displayType: .plus)
+
+        let controller: UIViewController
+        if FeatureFlag.patron.enabled, SubscriptionHelper.activeSubscriptionTier == .patron {
```
Comment (leandroalonso) [inline podcasts/Onboarding/Models/PlusPurchaseModel.swift]: A small suggestion for readability improvement:

```suggestion
        if FeatureFlag.patron.enabled, SubscriptionHelper.activeTier == .patron {
```

File: podcasts/Onboarding/Patron/UnlockHaptic.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        // Add some knocks to amplify intensity
+        addKnocks()
+
+        // 🏃‍♀️ You start walking
+        addMediumHeartBeats()
+
+        // 🏃‍♂️🏃‍♂️ You're jogging at a brisk pace
+        addFastHeartBeats()
+
+        // I AM SPRINTING AS FAST AS I CAN 🏃‍♂️🏃‍♂️🏃‍♂️🏃‍♂️💨💨💨
```
Comment (leandroalonso) [inline podcasts/Onboarding/Patron/UnlockHaptic.swift:50]: 😂

**Review by leandroalonso — APPROVED** [review]
@emilylaguna fantastic work on that! I tested on an actual device, iPad, and the smallest devices with iOS 15. All works as expected!

File: podcasts/Onboarding/Models/PlusPurchaseModel.swift (hunk 1 above)
Comment (emilylaguna) [reply podcasts/Onboarding/Models/PlusPurchaseModel.swift]: Done here: 14e98944ed09bc2c65d24e64a386c22ae67b73f9

**Pull request merged**